x/concentrated-liquidity/genesis.go

```diff
@@ -184,15 +194,34 @@ func (k Keeper) ExportGenesis(ctx sdk.Context) *genesis.GenesisState {
 		if err != nil {
 			panic(err)
 		}
+
 		feeAccumPositionRecord, err := feeAccumObject.GetPosition(feePositionKey)
 		if err != nil {
 			panic(err)
 		}
 
+		// Retrieve uptime incentive accumulator state for position
+		positionName := string(types.KeyPositionId(position.PositionId))
+		uptimeAccumulators, err := k.GetUptimeAccumulators(ctx, position.PoolId)
+		if err != nil {
+			panic(err)
+		}
```

Comment:
What we could do is, store this var outside the for loop as well as a var that says what the last pool ID uptime accumulator that was retrieved was. This way, if we have many positions in the same pool, we don't have to keep retrieving the same uptime accum.

Reply:
hmm, have we done something similar elsewhere in the code? just asking for reference

Reply:
What I mean is, you can set a var outside this for loop that is called something like `currentPoolId` and set it to zero as well as `currentPoolUptimeAccumulators`, then, on each iteration, set the `currentPoolId` var to whatever the poolId is for that iteration as well as store the `currentPoolUptimeAccumulators` outside of the for loop. Then, if the pool ID matched twice in a row or more, we wouldn't fetch the uptime accumulator multiple times which I imagine would improve performance when doing a state export.

That being said its not at all blocking, as I don't really know how much better in terms of performance this would be.